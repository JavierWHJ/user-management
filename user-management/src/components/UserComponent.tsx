import React from 'react';
import { Users } from '../type';

interface Props {
    user: Users | undefined
}

const UserComponent = (props: Props) => {
    return (
        <div>
            {props.user?.firstName}
        </div>
    );
}

export default UserComponent;
