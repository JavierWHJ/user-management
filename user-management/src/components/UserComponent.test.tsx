import React from "react";
import UserComponent from "./UserComponent";

import { render } from "@testing-library/react";

const user = {
    firstName: "Bob",
    lastName: "Doe",
    email: "bob.doe@gmail.com",
    dob: 785635200000,
    id: 3
  };

it("matches snapshot", () => {
  const { asFragment } = render(<UserComponent user={user} onSelectUserDelete={() => console.log()} onSelectUserUpdate={() => console.log()} />);

  expect(asFragment()).toMatchSnapshot();
});