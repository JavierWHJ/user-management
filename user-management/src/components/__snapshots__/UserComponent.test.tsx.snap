// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <tr>
    <td>
      <div
        class="d-flex align-items-center"
      >
        <div
          class="avatar-circle"
        >
          <span
            class="initials"
          >
            BD
          </span>
        </div>
        <span
          class="ml-3"
        >
          Bob Doe
        </span>
      </div>
    </td>
    <td>
      bob.doe@gmail.com
    </td>
    <td>
      24 Nov 1994
    </td>
    <td>
      <div
        class="d-flex justify-content-center mt-2"
      >
        <button
          class="mr-4 btn btn-success btn-block"
          type="button"
        >
          Update
        </button>
        <button
          class="mt-0 btn btn-danger btn-block"
          type="button"
        >
          Delete
        </button>
      </div>
    </td>
  </tr>
</DocumentFragment>
`;
